front-project/src/components/signUpForm.component.js

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import { Form, Field } from "react-final-form";
+import Avatar from "@mui/material/Avatar";
+import Button from "@mui/material/Button";
+import CssBaseline from "@mui/material/CssBaseline";
+import TextField from "@mui/material/TextField";
+import Link from "@mui/material/Link";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
+import Typography from "@mui/material/Typography";
+import Container from "@mui/material/Container";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import { Link as RouterLink } from "react-router-dom";
+
+const TextFieldAdapter = ({ input, meta, ...rest }) => {
+  const visibleError = !meta.modifiedSinceLastSubmit && meta.submitError;
+
+  return (
+    <TextField
+      {...input}
+      {...rest}
+      error={Boolean(visibleError)}
+      helperText={visibleError}
+    />
+  );
+};
+
+const theme = createTheme();
+
+export default function SignUpForm({ toggleFormType }) {
```

Comment:
i think you don't need to create theme in each component, just wrap root component in ThemeProvider
learn please how theming works in mui - https://mui.com/material-ui/customization/theming